src/wallets/sendTransaction/SendAdvancedTransaction.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {FunctionComponent} from 'react'
+import {ModalProps} from 'antd/lib/modal'
+import {ModalLocker} from '../../common/MantisModal'
+import {Dialog} from '../../common/Dialog'
+import {DialogInput, DialogTextArea} from '../../common/dialog/DialogInput'
+import {DialogColumns} from '../../common/dialog/DialogColumns'
+import {Wei} from '../../common/units'
+import {DialogAddressSelect} from '../../address-book/DialogAddressSelect'
+import {FeeEstimates} from '../../common/wallet-state'
+import {useTranslation} from '../../settings-state'
+import {AdvancedTransactionParams} from './common'
+
+interface SendAdvancedTransactionProps {
+  onSend: () => void
+  availableAmount: Wei
+  transactionParams: AdvancedTransactionParams
+  setTransactionParams: (advancedParams: Partial<AdvancedTransactionParams>) => void
+  estimateTransactionFee: () => Promise<FeeEstimates>
+  onCancel: () => void
+}
+
+export const SendAdvancedTransaction: FunctionComponent<SendAdvancedTransactionProps &
+  ModalProps> = ({
+  transactionParams,
+  setTransactionParams,
+  onCancel,
+  onSend,
+}: SendAdvancedTransactionProps & ModalProps) => {
+  const {t} = useTranslation()
+  const modalLocker = ModalLocker.useContainer()
+
+  const {amount, recipient, gasLimit, gasPrice, data, nonce} = transactionParams
+
+  return (
+    <>
+      <Dialog
+        title={t(['wallet', 'title', 'send'])}
+        leftButtonProps={{
+          onClick: onCancel,
+          disabled: modalLocker.isLocked,
+        }}
+        rightButtonProps={{
+          children: t(['wallet', 'button', 'sendTransaction']),
+          onClick: onSend,
+        }}
+        onSetLoading={modalLocker.setLocked}
+        type="dark"
+      >
+        <DialogAddressSelect
+          setRecipient={(recipient) => setTransactionParams({recipient})}
+          recipient={recipient}
+        />
+        <DialogInput
+          label={t(['wallet', 'label', 'amount'])}
+          id="tx-amount"
+          onChange={(e): void => setTransactionParams({amount: e.target.value})}
+          formItem={{
+            name: 'tx-amount',
+            initialValue: amount,
+          }}
+        />
+        <DialogColumns>
+          <DialogInput
+            label={t(['wallet', 'label', 'gasLimit'])}
+            id="tx-gas-limit"
+            onChange={(e): void => setTransactionParams({gasLimit: e.target.value})}
+            formItem={{
+              name: 'tx-gas-limit',
+              initialValue: gasLimit,
+            }}
+          />
+          <DialogInput
+            label={t(['wallet', 'label', 'gasPrice'])}
+            id="tx-gas-price"
+            onChange={(e): void => setTransactionParams({gasPrice: e.target.value})}
+            formItem={{
+              name: 'tx-gas-price',
+              initialValue: gasPrice,
+            }}
+          />
+        </DialogColumns>
+        <DialogTextArea
+          label={t(['wallet', 'label', 'data'])}
```

Comment:
We should disable newlines in the textarea. It should be only "multiline" to fit the content, but otherwise newline doesn't make sense for the data param.

Reply:
Is validation enough for this or do you want to automatically strip input?

Reply:
It's OK as it is. :+1: